File: week-1-c/Assignments/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // initiate height variable.
    int height;

    // Repeat question to user untill value is greater than 1.
    do
    {
        height = get_int("Enter height: ");
    }
    while (height < 1);

    // initiate spaces and block variables.
    int spaces = (height - 1);
    int blocks = 1;

    // While i is less than height do the following.
    for (int i = 0; i < height; i++)
    {
        // initiate j with the current value of spaces, print j amount of spaces, and take 1 away
        // from spaces.
        for (int j = spaces; j > 0; j--)
        {
            printf(" ");
        }
        spaces--;

        // initiate k with the current value of blocks, print k amount of blocks, and take 1 away
        // from blocks.
        for (int k = blocks; k > 0; k--)
        {
            printf("#");
        }
        blocks++;

        printf("\n");
    }
}

File: temp-folder/Assignments/sentimental-mario-more/mario.py
from cs50 import get_int
# Check if users input is greater than 0 and less than 8.
while True:
    height = get_int("Height: ")
    if height > 0 and height <= 8:
        break

# Set variables based on users input.
width = height - 1
block = 1

# Loop through the height while printing the spaces and blocks.
for _ in range(height):
    print(" " * width, end="")
    width -= 1
    print("#" * block, end="  ")
    print("#" * block)
    block += 1
